### PR TITLE
Series of smaller (mostly) diagnostics fixes

### PR DESCRIPTION
**Pull request by rsmmr**

### CODE REVIEW

**Comment by duffy-corelight** [comment]
Fixing precedence of “in” will close https://github.com/zeek/spicy/pull/293

File: hilti/src/ast/node.cc [hunk 1]
```diff
@@ -65,6 +65,9 @@ std::string Node::render(bool include_location) const {
             s += util::fmt(" (cxx-id: %s)", *cppid);
     }
 
+    if ( auto e = this->tryAs<Expression>() )
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply hilti/src/ast/node.cc:68]: ack

File: hilti/src/compiler/visitors/validator.cc [hunk 2]
```diff
@@ -387,6 +387,11 @@ struct Visitor : public visitor::PostOrder<void, Visitor> {
 
             if ( f.isStatic() && f.default_() )
                 error("&default is currently not supported for static fields", p);
+
+            if ( auto d = f.default_() ) {
+                if ( d->type() != f.type() )
+                    error(fmt("type mismatch for &default expression, expecting type %s", f.type()), p);
+            }
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply hilti/src/compiler/visitors/validator.cc:399]: changed

File: hilti/src/compiler/visitors/scope-builder.cc [hunk 3]
```diff
@@ -162,7 +162,10 @@ struct VisitorPass1 : public visitor::PostOrder<void, VisitorPass1> {
             if ( lc.input().type().template isA<type::Unknown>() )
                 return lc.input().type();
 
-            return lc.input().type().iteratorType(true).dereferencedType();
+            if ( auto t = lc.input().type(); type::isIterable(t) )
+                return t.iteratorType(true).dereferencedType();
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply hilti/src/compiler/visitors/scope-builder.cc:166]: agree, but will need a larger change as there are more properties using this pattern, so postponing

**Pull request merged**